controllers/nutanixcluster_controller.go

```diff
@@ -308,17 +308,25 @@ func (r *NutanixClusterReconciler) reconcileCredentialRefDelete(ctx context.Cont
 	}
 	err = r.Client.Get(ctx, secretKey, secret)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			log.V(1).Info(fmt.Sprintf("Secret %s in namespace %s for cluster %s not found. Ignoring since object must be deleted", secret.Name, secret.Namespace, nutanixCluster.Name))
+			return nil
+		}
 		return err
 	}
 	ctrlutil.RemoveFinalizer(secret, infrav1.NutanixClusterCredentialFinalizer)
 	log.V(1).Info(fmt.Sprintf("removing finalizers from secret %s in namespace %s for cluster %s", secret.Name, secret.Namespace, nutanixCluster.Name))
 	if err := r.Client.Update(ctx, secret); err != nil {
 		return err
 	}
-	log.Info(fmt.Sprintf("removing secret %s in namespace %s for cluster %s", secret.Name, secret.Namespace, nutanixCluster.Name))
-	if err := r.Client.Delete(ctx, secret); err != nil {
-		return err
+
+	if secret.DeletionTimestamp.IsZero() {
+		log.Info(fmt.Sprintf("removing secret %s in namespace %s for cluster %s", secret.Name, secret.Namespace, nutanixCluster.Name))
+		if err := r.Client.Delete(ctx, secret); err != nil {
```

Comment:
should we update the conditions too with this state

Reply:
should we be deleting secret by ourselves? per https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/ see eg of pv-protection. we are not the owner controller of secret resource cc @yannickstruyf3

Reply:
I dont think we should be calling delete on secret in our operator. https://martinheinz.dev/blog/74 we should remove finalizer, update it on secret and continue. let the operator which handles secret object take care of it. we should not call delete inside deletetimestamp = 0 case as its already being deleted anyway

Reply:
Remove the secret deletion for now and only cleanup finalizer

Reply:
Other CAPI providers usually delete the secret (if using a secret):
 - CAPV: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/controllers/vspherecluster_reconciler.go#L213
 - CAPH: https://github.com/syself/cluster-api-provider-hetzner/blob/1b1b4c55071aaf9001317649df23e585118e4478/controllers/hetznercluster_controller.go#L264C34-L264C34

Also in CAPI test framework there is expectation that after cluster delete will be no resources in namespace:
https://github.com/kubernetes-sigs/cluster-api/blob/main/test/framework/cluster_helpers.go#L237